src/dbnode/persist/types.go

```diff
@@ -322,12 +320,49 @@ const (
 	FileSetIndexContentType
 )
 
+// SeriesMetadataLifeTime describes the memory life time type.
+type SeriesMetadataLifeTime int
+
+const (
+	// SeriesMetadataLifeTimeLongLived means the underlying memory's life time is long lived and exceeds
+	// the execution duration of the series metadata receiver.
+	SeriesMetadataLifeTimeLongLived SeriesMetadataLifeTime = iota
+	// SeriesMetadataLifeTimeShortLived means that the underlying memory is only valid for the duration
+	// of the OnFlushNewSeries call. Must clone the underlying bytes in order to extend the life time.
+	SeriesMetadataLifeTimeShortLived
+)
+
+// SeriesMetadataType describes the type of series metadata.
+type SeriesMetadataType int
+
+const (
+	// SeriesDocumentType means the metadata is in doc.Document form.
+	SeriesDocumentType SeriesMetadataType = iota
+	// SeriesIDAndEncodedTagsType means the metadata is in IDAndEncodedTags form.
+	SeriesIDAndEncodedTagsType
+)
+
+// IDAndEncodedTags contains a series ID and encoded tags.
+type IDAndEncodedTags struct {
+	ID          []byte
+	EncodedTags []byte
```

Comment:
Also, might want to use `ident.BytesID`/ `ts.EncodedTags` type aliases here instead of `[]byte`.